### PR TITLE
Add script to create permalink pages

### PR DESCRIPTION
**Pull request by mdo**
_This isn't the best work I've done, but it's a start that I'm damned proud of lol. And it works! 😄 _

This PR adds a script that loops through the `icons/` directory, checks for an existing associated permalink page in `docs/content/icons/`, and if one does not exist, then creates said file with a front matter template. I know nothing about writing JS still, so this was pieced together with countless Googling and what not, so all feedback is welcome.

---

**Reference art:**
- https://stackoverflow.com/questions/32511789/looping-through-files-in-a-folder-node-js/32511953
- https://stackoverflow.com/questions/31195391/create-a-file-if-it-doesnt-already-exist
- https://nodejs.dev/writing-files-with-nodejs

### CODE REVIEW

File: build/build-pages.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+
+const iconsDir = path.join(__dirname, '../icons/')
+const pagesDir = path.join(__dirname, '../docs/content/icons/')
+
+function capitalizeFirstLetter(string) {
```
Comment (Johann-S) [inline build/build-pages.js]: arrow function here, should be enough

File: build/build-pages.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+(async () => {
+  try {
+    const files = await fs.promises.readdir(iconsDir)
+
+    for (const file of files) {
+      const iconBasename = path.basename(file, path.extname(file))
+      const iconTitleCap = capitalizeFirstLetter(iconBasename)
+      const iconTitle = iconTitleCap.split('-').join(' ')
+      const pageName = path.join(pagesDir, iconBasename + '.md')
```
Comment (Johann-S) [inline build/build-pages.js]: ES6 template to concat string

File: build/build-pages.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env node
+
+'use strict'
+
+const fs = require('fs')
```
Comment (Johann-S) [inline build/build-pages.js]: I think we should consider using promises as much as possible: `require('fs').promises`

File: build/build-pages.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      const pageName = path.join(pagesDir, iconBasename + '.md')
+
+      const pageTemplate = `---
+title: ${iconTitle}
+layout: icon
+categories:
+tags:
+---\n`
+
+      fs.access(pageName, fs.F_OK, err => {
```
Comment (Johann-S) [inline build/build-pages.js]: using promisified version of `access`

File: build/build-pages.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        if (err) {
+          fs.writeFile(pageName, pageTemplate, err => {
```
Comment (Johann-S) [inline build/build-pages.js]: using promisified version of `writeFile`

**Comment by XhmikosR** [comment]
@Johann-S can you take care of your comments? And make sure we handle all Promise errors everywhere please? ❤

BTW we also have #50 which can still throw an unhandled Promise rejection.

**Comment by mdo** [comment]
I'd like to take a crack at a few things if that's okay :).

**Comment by mdo** [comment]
Didn't get much traction on this part myself—definitely still lost on a lot of the JS side :). Happy to defer to someone or accept this as-is even.

**Comment by XhmikosR** [comment]
Unless @Johann-S has time to take care of his suggestions, this works, even if it's not perfect. 🙂

File: build/build-pages.js (hunk 1 above)
Comment (XhmikosR) [reply build/build-pages.js:11]: A function is fine too also, we don't enforce function expressions.

**Comment by mdo** [comment]
Merging for now, but I'd still love any updates obviously :).

**Pull request merged**